### PR TITLE
Be/update/project setup and authentification

### PR DESCRIPTION
**Pull request by xAmiBa**
- Flask app created and configured with db-connection
- roammates & test_roammates databases created and connected
- repository classes created
- model classes created
- token generator function an JWT configured

- 100% testing:
Screenshot 2023-12-13 at 17 08 46

- NOTICE: to install databases locally follow api/recipie.md

### CODE REVIEW

File: api/lib/Preference.py
Comment (PiotrSurowiec90) [inline api/lib/Preference.py:1]: Can we Use dataclass for models [python dataclass](https://docs.python.org/3/library/dataclasses.html)

Can We have comment description of what the models represents, what is the relations between ths model ad other models and fields description if needed( Imagine someone who doesn't know anything about the project is going to read it.)

File: api/lib/Profile.py
Comment (PiotrSurowiec90) [inline api/lib/Profile.py:1]: Can we Use dataclass for models [python dataclass](https://docs.python.org/3/library/dataclasses.html)

Can We have comment description of what the models represents, what is the relations between other table and field description if needed( Imagine someone who doesn't know anything about the project is going to read it.)

File: api/lib/Request.py
Comment (PiotrSurowiec90) [inline api/lib/Request.py:1]: Can we Use dataclass for models [python dataclass](https://docs.python.org/3/library/dataclasses.html)

Can We have comment description of what the models represents, what is the relations between other table and field description if needed( Imagine someone who doesn't know anything about the project is going to read it.)

File: api/lib/User.py
Comment (PiotrSurowiec90) [inline api/lib/User.py:1]: Can we Use dataclass for models [python dataclass](https://docs.python.org/3/library/dataclasses.html)

Can We have comment description of what the models represents, what is the relations between other table and field description if needed( Imagine someone who doesn't know anything about the project is going to read it.)

File: api/lib/database_connection.py
Comment (PiotrSurowiec90) [inline api/lib/database_connection.py:1]: Can database name be stored as env var, i suggest this library  - [dotenv](https://pypi.org/project/python-dotenv/)

File: api/seeds/roammates_seed.sql
Comment (PiotrSurowiec90) [inline api/seeds/roammates_seed.sql:1]: - User email needs unique constraint [UNIQUE](https://www.w3schools.com/sql/sql_unique.asp)
- FK request from should be present ( request table)
- All the dropdown fields (ageslot, gender, season, category, continent) to accept choices.
One way to do it is:
`CREATE TABLE YourTable (
    YourTextField VARCHAR(255) CHECK (YourTextField IN ('Choice1', 'Choice2', 'Choice3'))
);
`

**Review by PiotrSurowiec90 — CHANGES_REQUESTED** [review]

File: api/lib/Preference.py
Comment (xAmiBa) [reply api/lib/Preference.py:1]: done

File: api/lib/Profile.py
Comment (xAmiBa) [reply api/lib/Profile.py:1]: done

File: api/lib/Request.py
Comment (xAmiBa) [reply api/lib/Request.py:1]: done

File: api/lib/User.py
Comment (xAmiBa) [reply api/lib/User.py:1]: done

File: api/lib/database_connection.py
Comment (xAmiBa) [reply api/lib/database_connection.py:1]: done, great suggestion!

File: api/seeds/roammates_seed.sql
Comment (xAmiBa) [reply api/seeds/roammates_seed.sql:1]: done!

**Review by PiotrSurowiec90 — APPROVED** [review]
Good JOB!

**Pull request merged**